src/tasks/TasksForm.js

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect } from "react";
+import TextField from "@material-ui/core/TextField";
+import Grid from "@material-ui/core/Grid";
+import DatePicker from "../components/DatePicker";
+import { useForm, Form } from "../components/useForm";
+import Button from "../components/Button";
+import Priority from "../components/Priority";
+import Subject from "../components/Subject";
+import OpenLayersMap from "../components/OpenLayersMap";
+
+const initialValues = {
+  id: 0,
+  taskName: "",
+  taskSubject: "",
+  myPriority: 5,
+  myDate: new Date(),
+  taskLocation: [0, 0],
+  isCompleted: 0,
+};
+
+export default function TasksForm({ addOrEdit, recordForEdit }) {
+  const { values, setValues, handleInputChange, resetForm } = useForm(
+    initialValues,
+    true
+  );
+
+  const handleSubmit = (e) => {
+    addOrEdit(values, resetForm);
+    resetForm();
+  };
+
+  const onClick = (subject) => setValues({ ...values, taskSubject: subject });
```

Comment:
`onChangeSubject`.  I also think you can use `handleInputChange `

Reply:
i get error e.target is undefind